docs/meson.build

```diff
@@ -1,16 +1,66 @@
-gnome.gtkdoc(
-  'fwupd',
-  src_dir : [
-    join_paths(meson.source_root(), 'libfwupd'),
-    join_paths(meson.source_root(), 'libfwupdplugin'),
-    join_paths(meson.build_root(), 'libfwupd'),
-    join_paths(meson.build_root(), 'libfwupdplugin'),
-    join_paths(meson.current_source_dir()),
+toml_conf = configuration_data()
+toml_conf.set('version', meson.project_version())
+
+fwupd_toml = configure_file(
+  input: 'fwupd.toml.in',
+  output: 'fwupd.toml',
+  configuration: toml_conf
+)
+
+fwupdplugin_toml = configure_file(
+  input: 'fwupdplugin.toml.in',
+  output: 'fwupdplugin.toml',
+  configuration: toml_conf
+)
+
+gidocgen = find_program('gi-docgen')
+
+custom_target('doc-fwupd',
+  input: [
+    fwupd_toml,
+    fwupd_gir[0],
+  ],
+  output: 'libfwupd',
+  command: [
+    gidocgen,
```

Comment:
shouldn't this be `gidocgen_dep` or similar?

Reply:
Nah, `gidocgen_dep` is the .pc file target, `gidocgen` is the binary. The former is only really needed to create the subproject.

Reply:
> But how are you going to find the binary if you didn't have it pre-installed

No idea; I don't have it locally so I guess it's finding it from the subpackage somehow.

Reply:
Fixed.